src/mode/mode_struct.rs

```diff
@@ -1215,10 +1270,43 @@ impl<T: Transformation> Mode<T> {
         } else {
             pepped_up_control_value.calc_distance_from(current_target_value.to_continuous_value())
         };
+
+        // If there is no previous pepped-up control value, set one
+        let previous_pepped_up_control_value = match self.state.previous_pepped_up_control_value {
+            None => pepped_up_control_value,
+            Some(value) => value,
+        };
+        self.state.previous_pepped_up_control_value = Some(pepped_up_control_value);
+
+        let time_expired = match self.state.previous_control_value_time {
+            None => true,
+            Some(time) => time.elapsed().as_millis() > CONTROL_MOVE_TIMEOUT,
```

Comment:
I think best practice would be to define `CONTROL_MOVE_TIMEOUT` as a `Duration` constant and then remove `as_millis()` (compare the durations directly).